### PR TITLE
Generate documentation with mdBook

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

File: docs/src/build/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## Building lockc
+
+The first step to try out lockc is to build it. There are two ways to do
+that:
+
+- **[Containerized]** build project binaries within container
+- **[Meson]** build binaries with Meson
```
Comment (vadorovsky) [inline docs/src/build/README.md:7]:
```suggestion
- **[Containerized]** - build project binaries within container
- **[Meson]** - build binaries with Meson
```

File: docs/src/configuration.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Getting started
+
+- [Build] How to build lockc from the sources
+- [Use] Configuring and using lockc locally
+- [Provision] Using Terraform for provisioning lockc
```
Comment (vadorovsky) [inline docs/src/configuration.md:5]:
```suggestion
- [Build] - How to build lockc from the sources
- [Use] - Configuring and using lockc locally
- [Provision] - Using Terraform for provisioning lockc
```

File: docs/src/SUMMARY.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- [Introduction](README.md)
+- [Architecture](architecture.md)
+- [Getting started](configuration.md)
+    - [Build](build/README.md)
+        - [Containerized](build/containerized.md)
+        - [Meson](build/meson.md)
+    - [Use](use/README.md)
+    - [Provision](terraform/README.md)
+        - [Base image](terraform/base-image.md)
+        - [Custom kernel](terraform/custom-kernel.md)
```
Comment (vadorovsky) [inline docs/src/SUMMARY.md:12]:
```suggestion
        - [(Optional) Custom kernel](terraform/custom-kernel.md)
```

File: docs/src/terraform/start-vm.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+terraform apply
+```
+
+ Terraform finished successfully, you should see the output with IP
+addresses of virtual machines, like:
+
+```bash
+ip_control_planes = {
+  "lockc-control-plane-0" = "10.16.0.225"
+}
```
Comment (vadorovsky) [inline docs/src/terraform/start-vm.md:33]: You need to close the code snippet here in this line, it doesn't render properly - https://mjura.github.io/lockc/terraform/start-vm.html

**Review by vadorovsky — CHANGES_REQUESTED** [review]
Good job! I found only one rendering error and have few suggestions.

File: docs/src/SUMMARY.md (hunk 3 above)
Comment (mjura) [reply docs/src/SUMMARY.md]: done

File: docs/src/build/README.md (hunk 1 above)
Comment (mjura) [reply docs/src/build/README.md]: done

File: docs/src/configuration.md (hunk 2 above)
Comment (mjura) [reply docs/src/configuration.md]: done

File: docs/src/terraform/start-vm.md (hunk 4 above)
Comment (mjura) [reply docs/src/terraform/start-vm.md:33]: Interesting that `mdbook test` didn't find this problem, during CI run

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -7,11 +7,18 @@ on:
...
       run: ./containerized-build.sh build
     - name: Lint
       run: ./containerized-build.sh lint
+    - name: Build book
+      run: ./scripts/mdbook.sh
+    - name: Deploy website
+      uses: JamesIves/github-pages-deploy-action@4.1.4
+      with:
+        branch: gh-pages
+        folder: docs/book
```
Comment (vadorovsky) [inline .github/workflows/rust.yml:24]: Keeping this job in this file might cause one problem.

At the top we have:

```
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
```

Which means that every step defined in this file is running:

* when a commit is pushed to the `main` branch - which happens when a PR is merged - I think that's exactly when we want to deploy docs :)
* when a PR is submitted - and in this case, I think that we **don't** want to deploy the website

So I think that we should do two things.

First one, keep only building the docs in this file, without deploying:

```suggestion
    - name: Build book
      run: ./scripts/mdbook.sh
```

And then create a separate file, i.e. `.github/workflows/mdbook.yml`, with the following content:

```
name: mdBook

on:
  push:
    branches: [ main ]

jobs:
  build-and-deploy:
    runs-on: ubuntu-latest
    steps:
    - name: Build book
      run: ./scripts/mdbook.sh
    - name: Deploy website
      uses: JamesIves/github-pages-deploy-action@4.1.4
      with:
        branch: gh-pages
        folder: docs/book
```

I think that should work. And also make this pull request green. Could you try it?

File: src/lib.rs [hunk 6]
```diff
@@ -1,342 +1,3 @@
...
-//! The main technology behind lockc is [eBPF](https://ebpf.io/) - to be more
-//! precise, its ability to attach to [LSM hooks](https://www.kernel.org/doc/html/latest/bpf/bpf_lsm.html)
-//!
-//! Please note that currently lockc is an experimental project, not meant for
-//! production environments. Currently we don't publish any official binaries or
-//! packages to use, except of a Rust crate. Currently the most convenient way
-//! to use it is to use the source code and follow the guide.
-//!
-//! If you need help or want to talk with contributors, please come chat with
-//! us on `#lockc` channel on the [Rust Cloud Native Discord server](https://discord.gg/799cmsYB4q).
```
Comment (vadorovsky) [inline src/lib.rs:13]: I would keep the old comment until this moment, as a short introduction in https://docs.rs/lockc/0.1.0/lockc/ and maybe add something like this at the beginning:

```
//! This is an auto-generated code documentation. For more detailed documentation
//! (with all information about usage, deployment and architecture) please check
//! out [the book](https://rancher-sandbox.github.io/lockc/).
```

**Review by vadorovsky — CHANGES_REQUESTED** [review]
2 more comments

File: docs/src/terraform/start-vm.md (hunk 4 above)
Comment (vadorovsky) [reply docs/src/terraform/start-vm.md:33]: Yes, looks like `mdbook test` is quite inefficient. Using something like https://github.com/markdownlint/markdownlint could be cool in the future.

File: src/lib.rs (hunk 6 above)
Comment (mjura) [reply src/lib.rs:13]: done

File: .github/workflows/rust.yml (hunk 5 above)
Comment (mjura) [reply .github/workflows/rust.yml]: yes, it will be more reasonable solution

**Review by vadorovsky — APPROVED** [review]

**Pull request merged**